### PR TITLE
get a user by id with resolved catalog

### PR DESCRIPTION
**Pull request by dpopp07**
Adds a route in REST and GraphQL for getting a single user by their DB generated ID. If they have a catalog, it will return all the full drink objects, not a list of drink IDs.

Todo:
- [x] ~Update OpenAPI doc~ Get REST outta there
- [x] Write tests

### CODE REVIEW

**Comment by dpopp07** [comment]
Ready for review

File: bartop-api/src/api/user/user.logic.js [hunk 1]
```diff
@@ -1,19 +1,35 @@
...
+    return await r.table(TABLE_NAME);
+  };
+
+  // get a user by the db-generated id
+  // will be `null` if user does not exist
+  const get = async id => {
+    const user = await r.table(TABLE_NAME).get(id);
+    // resolve drink objects in the catalog
+    if (user && user.catalog) {
+      const drinks = await r.table('drinks').getAll(...user.catalog);
```
Comment (jamesvclements) [inline bartop-api/src/api/user/user.logic.js:28]: clean 🚿

File: bartop-api/src/utils/errorCreator.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module.exports.notFound = id => {
+  const newError = new Error();
+  newError.name = 'ResourceNotFoundError';
+  newError.message = `Resource with ID:${id} does not exist.`;
```
Comment (jamesvclements) [inline bartop-api/src/utils/errorCreator.js:4]: Nit picky but would you like to add a space between `ID:` and `${id}`

Comment (jamesvclements) [reply bartop-api/src/utils/errorCreator.js:4]: Also if you wanted you could pass the `message` in the constructor but I kinda like that the `name` is assigned before the `message`

Comment (dpopp07) [reply bartop-api/src/utils/errorCreator.js:4]: did that in the other PR actually

File: bartop-api/test/utils/testObjects/user/index.js [hunk 3]
```diff
@@ -14,7 +14,14 @@ const user3 = {
   auth0Id: 'billyd@ufl.edu|9164372819546791'
 };
 
+const user4 = {
+  auth0Id: 'peteforpres@midearth.com|2461849122549183'
```
Comment (jamesvclements) [inline bartop-api/test/utils/testObjects/user/index.js:18]: 💀💀💀

**Review by jamesvclements — APPROVED** [review]
buenisimo

**Pull request merged**